doc/examples/applications/plot_coins_segmentation.py

```diff
@@ -35,7 +35,9 @@
 
 """
 
-fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3))
+fig = plt.figure(figsize=(6,3))
+ax1 = fig.add_subplot(1, 2, 1, adjustable='box-forced')
+ax2 = fig.add_subplot(1, 2, 2, sharex = ax1, sharey = ax1, adjustable='box-forced')
```

Comment:
Ok. I'll change these back to 
```python
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3), sharex=True, sharey=True, subplot_kw={'adjustable':'box-forced'} )
```
What about the specific cases wher plt.subplots does not seem flexible enough? There are two specific cases:
* sharing axes across figures
* sharing only specific axes in a figure - for example the first column